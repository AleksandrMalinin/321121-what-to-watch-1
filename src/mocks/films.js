/* eslint camelcase: ["error", {properties: "never"}] */

export const films = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedies`
  },
  {
    title: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`
  },
  {
    title: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Documentary`
  },
  {
    title: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Dramas`
  },
  {
    title: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Horror`
  }
];
