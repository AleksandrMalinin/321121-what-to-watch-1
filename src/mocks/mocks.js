/* eslint camelcase: ["error", {properties: "never"}] */

export const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    preview_image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedies`
  },
  {
    name: `Bohemian Rhapsody`,
    preview_image: `img/bohemian-rhapsody.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`
  },
  {
    name: `Macbeth`,
    preview_image: `img/macbeth.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Documentary`
  },
  {
    name: `Aviator`,
    preview_image: `img/aviator.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Dramas`
  },
  {
    name: `We need to talk about Kevin`,
    preview_image: `img/we-need-to-talk-about-kevin.jpg`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Horror`
  }
];

export const defaultMovie = {
  background_color: `#AD9F8B`,
  background_image: `https://es31-server.appspot.com/wtw/static/film/background/Dardjeeling_Limited.jpg`,
  description: `A year after their father's funeral, three brothers travel across India by train in an attempt to bond with each other.`,
  director: `Wes Anderson`,
  genre: `Adventure`,
  id: 1,
  is_favorite: false,
  name: `Dardjeeling Limited`,
  poster_image: `https://es31-server.appspot.com/wtw/static/film/poster/Dardjeeling_Limited.jpg`,
  preview_image: `https://es31-server.appspot.com/wtw/static/film/preview/dardjeeling_limited.jpg`,
  preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: 3.6,
  released: 2007,
  run_time: 91,
  scores_count: 165106,
  starring: [`Owen Wilson`, `Adrien Brody`, `Jason Schwartzman`],
  video_link: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`
};
