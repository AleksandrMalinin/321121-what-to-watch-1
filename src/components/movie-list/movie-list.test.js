import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import MovieList from './movie-list.jsx';
import {films} from '../../mocks/mocks.js';

it(`MovieList correctly renders`, () => {

  const tree = renderer
  .create(
      <BrowserRouter>
        <MovieList
          movies={films}
        />
      </BrowserRouter>)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
