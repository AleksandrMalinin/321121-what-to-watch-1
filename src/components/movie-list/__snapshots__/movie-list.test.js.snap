// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList correctly renders 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      className="small-movie-card__image"
      height="175"
      muted={true}
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="280"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        type="video/mp4"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      className="small-movie-card__image"
      height="175"
      muted={true}
      poster="img/bohemian-rhapsody.jpg"
      width="280"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        type="video/mp4"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      className="small-movie-card__image"
      height="175"
      muted={true}
      poster="img/macbeth.jpg"
      width="280"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        type="video/mp4"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      className="small-movie-card__image"
      height="175"
      muted={true}
      poster="img/aviator.jpg"
      width="280"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        type="video/mp4"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      className="small-movie-card__image"
      height="175"
      muted={true}
      poster="img/we-need-to-talk-about-kevin.jpg"
      width="280"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        type="video/mp4"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      />
    </h3>
  </article>
</div>
`;
